File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.java
class Solution {
    public int maxProfit(int[] prices) {
        int len = prices.length;
        int max_profit = 0;
        if(len == 1){
            return max_profit;
        }
        int buying_price = prices[0];
        int selling_price = prices[0];
        
        for(int i=1;i<len;i++){
           buying_price = Math.min(buying_price,prices[i]);
           max_profit = Math.max(max_profit,prices[i]-buying_price);
        }
        return max_profit; 
    }
}

File: Data Structures & Algorithms/trapping-rain-water/submission-1.js
class Solution {
    /**
     * @param {number[]} height
     * @return {number}
     */
    trap(height) {
        const length = height.length;
        let total = 0;
        if(length < 2){
            return total;
        }

        const leftMax = new Array(length).fill(0);
        leftMax[1] = height[0];
        const rightMax = new Array(length).fill(0);
        for(let i=2;i<length;i++){
            leftMax[i] = Math.max(height[i-1],leftMax[i-1]);
        }
        rightMax[length-2] = height[length-1];
        for(let i=length-3; i>=0; i--){
            rightMax[i] = Math.max(rightMax[i+1],height[i+1]);
        }

        let h = 0
        for(let i=0; i<length; i++){
            h = Math.min(leftMax[i],rightMax[i]);
            total += h > height[i] ? h - height[i] : 0;
        }
        return total;
    }
}
